tests/PHPUnit/Framework/Mock/TestConfig.php

```diff
@@ -18,25 +20,28 @@ class TestConfig extends Config
     private $isConfigTestEventPosted = false;
     private $doSetTestEnvironment = false;
 
-    public function __construct($pathGlobal = null, $pathLocal = null, $pathCommon = null, $allowSave = false, $doSetTestEnvironment = true)
+    public function __construct(GlobalSettingsProvider $settings, $allowSave = false, $doSetTestEnvironment = true,
```

Comment:
I would suggest breaking this constructor in many lines. Choose between having all in one line or each one in its own line. Only one param in its own line is messy.

```php
public function __construct(
    GlobalSettingsProvider $settings,
    $allowSave = false,
    $doSetTestEnvironment = true,
    TestingEnvironment $testingEnvironment = null
) {
}
```